github-pullrequest-plugin/src/main/java/com/github/kostyasha/github/integration/branch/events/impl/GitHubBranchRestrictionFilter.java

```diff
@@ -0,0 +1,127 @@
+package com.github.kostyasha.github.integration.branch.events.impl;
+
+import com.github.kostyasha.github.integration.branch.GitHubBranch;
+import com.github.kostyasha.github.integration.branch.GitHubBranchCause;
+import com.github.kostyasha.github.integration.branch.GitHubBranchRepository;
+import com.github.kostyasha.github.integration.branch.GitHubBranchTrigger;
+import com.github.kostyasha.github.integration.branch.events.GitHubBranchEvent;
+import com.github.kostyasha.github.integration.branch.events.GitHubBranchEventDescriptor;
+
+import hudson.Extension;
+import hudson.model.TaskListener;
+
+import org.kohsuke.github.GHBranch;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Set;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class GitHubBranchRestrictionFilter extends GitHubBranchEvent {
+
+    private static final String DISPLAY_NAME = "Branch Restrictions";
+
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+    private static final Logger LOGGER = LoggerFactory.getLogger(GitHubBranchRestrictionFilter.class);
+
+    private boolean exclude;
+
+    private boolean matchAsPattern;
+    private Set<String> matchCriteria;
+
+    @DataBoundConstructor
+    public GitHubBranchRestrictionFilter() {
+    }
+
+    public String getMatchCriteriaStr() {
+        return String.join(LINE_SEPARATOR, matchCriteria);
+    }
+
+    @DataBoundSetter
+    public void setMatchCriteriaStr(String matchCriteria) {
+        this.matchCriteria = Stream.of(matchCriteria
+                .split(LINE_SEPARATOR))
+                .collect(Collectors.toSet());
+    }
+
+    public boolean isExclude() {
+        return exclude;
+    }
+
+    @DataBoundSetter
+    public void setExclude(boolean exclude) {
+        this.exclude = exclude;
+    }
+
+    public boolean isMatchAsPattern() {
+        return matchAsPattern;
+    }
+
+    @DataBoundSetter
+    public void setMatchAsPattern(boolean matchAsPattern) {
+        this.matchAsPattern = matchAsPattern;
+    }
+
+    @Override
+    public GitHubBranchCause check(GitHubBranchTrigger gitHubBranchTrigger, GHBranch remoteBranch, GitHubBranch localBranch,
+                                   GitHubBranchRepository localRepo, TaskListener listener) {
+        String name = remoteBranch.getName();
+        if (matchCriteria.isEmpty() || branchIsAllowed(name)) {
+            if (matchCriteria.isEmpty()) {
+                LOGGER.warn("Branch restriction filter added but no match criteria set, all branches allowed");
+            }
+            return toCause(remoteBranch, localRepo, false, "Branch [%s] allowed by branch name restriction filter", name);
```

Comment:
@jgangemi seems here is bug. When it allowed to build, then it should return `null` to allow other checks. 
Either cause after upgrade triggers build. 

```
Running GitHub Branch trigger check for Nov 21, 2016 2:41:19 AM on KostyaSha-org/test-looper
GitHub rate limit before check: GHRateLimit{remaining=4982, limit=5000, resetDate=Mon Nov 21 03:35:07 MSK 2016}
Build of branch [for-pr] skipped: Branch [for-pr] filtered by branch name restriction filter.
Finished GitHub Branch trigger check for KostyaSha-org/test-looper at Nov 21, 2016 2:41:20 AM. Duration: 1349ms
```

```
	GitHub Branch master, Branch [master] allowed by branch name restriction filter
```
So filter should either return `null` to allow other checks either `skip=true` to ignore other checks.